### PR TITLE
Make it possible to delete a user and all of its data via API 

### LINKED ISSUE #600: Make it possible to delete a user and all of its data via API
**Problem**
We want to be able to delete a user record and all of its associated data. Eventually we'll add a UI to enable this, but to start off we should be able to do it safely and programmatically.

(This will probably require some database migrations that improve how associated records are deleted... at the moment some resources are not automatically deleted when a parent resource is attempted to be deleted.)

This will also be necessary for GDPR compliance.

**Solution**
Set up an API for `DELETE /api/users/:id` (or similar) which deletes a user and all of its data.

**Testing**
Write automated tests and verify that all the user data is deleted, and no other user is affected.

**Questions, or need help getting started?**
Feel free to ask below, or [ping us on Slack](https://join.slack.com/t/papercups-io/shared_invite/zt-h0c3fxmd-hZi1Zp8~D61S6GD16aMqmg) :)

(You can also check out our [CONTRIBUTING.md](https://github.com/papercups-io/papercups/blob/master/CONTRIBUTING.md))

### PR DESCRIPTION
**Pull request by fmterrorf**
### Description

- [x] DB migration to fix how associated resources are deleted when `user` is deleted.

### Issue

Closes #600 

## Checklist

- [x] Everything passes when running `mix test`
- [x] Ran `mix format`
- [x] No frontend compilation warnings

### CODE REVIEW

File: priv/repo/migrations/20210302011754_add_cascade_deletes_to_user_references.exs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    alter table(:google_authorizations) do
+      modify(:user_id, references(:users, on_delete: :delete_all))
+    end
+
+    alter table(:tags) do
+      modify(:creator_id, references(:users, on_delete: :nilify_all))
+    end
+
+    alter table(:conversation_tags) do
+      modify(:creator_id, references(:users, on_delete: :delete_all))
```
Comment (reichert621) [inline priv/repo/migrations/20210302011754_add_cascade_deletes_to_user_references.exs:40]: i think this one can just be `nilify_all` as well :+1:

**Review by reichert621 — COMMENTED** [review]
great work @fmterrorf! 🎉 

just one small thing and then it should be good to go :+1:

**Comment by a8t** [comment]
@fmterrorf I'm a novice with Elixir. Reading this PR helped a lot. May I ask you some simple questions about this PR to further help me learn?

**Comment by fmterrorf** [comment]
@a8t yeah for sure! I'd be happy answer any questions

**Review by reichert621 — APPROVED** [review]
great work @fmterrorf! thanks for the help on this 🎉 🚀

**Review by a8t — COMMENTED** [review]
Was this migration generated automatically somehow, or did you write it by hand?

File: test/chat_api/users_test.exs [hunk 2]
```diff
@@ -1,12 +1,45 @@
...
   import ChatApi.Factory
 
+  alias ChatApi.Repo
   alias ChatApi.Users
   alias ChatApi.Users.{User, UserProfile, UserSettings}
 
+
+  describe "user" do
+    setup do
+      {:ok, user: insert(:user)}
```
Comment (a8t) [inline test/chat_api/users_test.exs:14]: Where does the `insert` function come from?

**Comment by a8t** [comment]
I left 2 comments. Thanks Daven!

File: test/chat_api/users_test.exs (hunk 2 above)
Comment (fmterrorf) [reply test/chat_api/users_test.exs:14]: The insert function is coming from this https://github.com/papercups-io/papercups/blob/master/test/support/factory.ex 

In `user_test.exs` we have call `  import ChatApi.Factory`.  The `factory.ex` file calls uses [ExMachina.ecto](https://hexdocs.pm/ex_machina/ExMachina.Ecto.html) which gives you that `insert` function

**Comment by fmterrorf** [comment]
Was this migration generated automatically somehow, or did you write it by hand?

I wrote it by hand by creating the migration using [mix ecto.gen.migration (My migration name)](https://hexdocs.pm/ecto_sql/Mix.Tasks.Ecto.Gen.Migration.html), and adding the up/down migration

**Comment by a8t** [comment]
Thanks @fmterrorf. It was useful reference for my recent PR.

**Pull request merged**